File: ABC-221/D_AC.py
from typing import List, Tuple

N: int = 200010
ans: List[int] = [0] * N

n = int(input())
x: List[Tuple[int, int]] = []
for _ in range(n):
    a, b = map(int, input().split())
    x.append((a, 1))
    x.append((a + b, -1))

x = list(sorted(x, key=lambda x: x[0]))

cnt: int = 0
for i in range(len(x) - 1):
    cnt += x[i][1]
    ans[cnt] += x[i + 1][0] - x[i][0]

for i in range(n - 1):
    print(ans[i + 1], end=" ")
print(ans[n])

File: ABC-229/D_AC.cpp
#include <iostream>
#include <vector>

using Int = long long;
using namespace std;

int main() {
  string s;
  cin >> s;
  Int k;
  cin >> k;

  Int n = s.size();
  /* 累積和
    cnt[r]-cnt[l]によってs[l,r)の'.'の数がわかる
*/
  vector<Int> cnt(n + 1);
  for (int i = 0; i < n; ++i) {
    cnt[i + 1] = cnt[i];
    if (s[i] == '.')
      cnt[i + 1]++;
  }

  Int ans = 0;
  Int r = 0;

  for (Int l = 0; l < n; l++) {
    while (r <= n - 1 && cnt[r + 1] - cnt[l] <= k) {
      r++;
    }
    ans = max(ans,r-l);
  }
    cout<<ans<<endl;

  return 0;
}
